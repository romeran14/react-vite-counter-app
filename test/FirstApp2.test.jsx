import { getByTestId, render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"


describe('Pruebas firt app', () => { 
    const title = "hola soy goku"
    const subtitle = "hola soy raditz"
    test('debe hacer match con el snapshot', () => { 
        
        
        const { container } = render( <FirstApp title={title}></FirstApp>)
        expect( container).toMatchSnapshot()
     })

     test('debe hacer match con el snapshot', () => { 
        const { container } = render( <FirstApp title={title}></FirstApp>)
        screen.debug()
        expect(screen)
     })

     test('debe de mostrar el titulo en un h1', () => { 
        render( <FirstApp title={title}></FirstApp>)
        expect( screen.getByRole('heading',{level:1}))
      })

      test('debe mostrar el subtitulo enviado por props', () => { 
        render( <FirstApp subTitle={subtitle} title={title}></FirstApp>)
        expect( screen.getAllByText(subtitle).length).toBe(2)
      })

 })