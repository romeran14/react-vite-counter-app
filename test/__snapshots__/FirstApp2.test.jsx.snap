// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas firt app debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    hola soy goku
     
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;

exports[`Pruebas firt app should macth snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    hola soy goku
     
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
